core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -143,39 +144,53 @@ static class ManifestReadTask extends BaseFileScanTask implements DataTask {
     private final FileIO io;
     private final Map<Integer, PartitionSpec> specsById;
     private final ManifestFile manifest;
-    private final Schema schema;
+    private final Schema dataTableSchema;
+    private final Schema projectedSchema;
 
     ManifestReadTask(
         Table table,
         ManifestFile manifest,
-        Schema schema,
+        Schema projectedSchema,
         String schemaString,
         String specString,
         ResidualEvaluator residuals) {
       super(DataFiles.fromManifest(manifest), null, schemaString, specString, residuals);
       this.io = table.io();
       this.specsById = Maps.newHashMap(table.specs());
       this.manifest = manifest;
-      this.schema = schema;
+      this.dataTableSchema = table.schema();
+      this.projectedSchema = projectedSchema;
     }
 
     @Override
     public CloseableIterable<StructLike> rows() {
-      return CloseableIterable.transform(manifestEntries(), file -> (StructLike) file);
+      Set<Integer> projectedIds = TypeUtil.getProjectedIds(DataFile.READABLE_METRICS.type());
```

Comment:
Could you please add tests where we project only some subfields in the struct value?
I don't expect schema pruning to happen on the Iceberg side but it should work correctly.

```
SELECT readable_metrics['col1'].lower_bound, readable_metrics['col1'].upper_bound ...
```

Reply:
Added and actually found two bugs:

- When readable_metrics is selected, was projecting out the metrics needed to calculate it
- Need to add projection logic on struct itself (ie, the value of readable_metrics map), otherwise MetricsStruct becomes out of order.

Reply:
The test is located in "TestMetadataTableReadableMetrics", which was collapsed by github as its too big